src/openseadragon.js

```diff
@@ -2622,10 +2623,13 @@ function OpenSeadragon( options ){
          * keys and booleans as values.
          */
         setImageFormatsSupported: function(formats) {
+            //TODO: how to deal with this within the data pipeline?
```

Comment:
I think you would just provide (or not provide) the given convertor. Or provide a convertor that throws specific error message (if it is not too ugly). The truth is that we cannot differentiate this on the convertor level easily (all are `image`), but when I saw what this method _really_ does (basically nothing, the flag is respected only within selected protocol), I felt like either redesigning it, or removing.

Reply:
DZI just throws error if `Format` attribute is set. I think canvas API would throw some error too if it cannot be used, and IMHO something understandable too. IIF just in case you receive `preferredFormats` attribute checks which of these is supported and then tries to request that one. That's it. I do see a usecase for this format differentiation, just felt like it is really used just once at some special occasion. Why not move this then to the protocol itself?

Reply:
Yeah, it may be that we don't really need this functionality. I think it was mostly just meant as a way to help save users from themselves. You'd be surprised how many people don't understand what image formats the browser supports!

If we do remove it, we need to keep the APIs and add deprecation warnings to them. Also it should go in the changelog.